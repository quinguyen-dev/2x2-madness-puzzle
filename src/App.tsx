import { createBrowserRouter, RouterProvider } from "react-router-dom";
import { Configuration, Game } from "./pages/index.ts";

const router = createBrowserRouter([
  { path: "/", Component: Configuration },
  { path: "game", Component: Game },
]);

export default function App() {
  return (
    <main className="h-screen flex justify-center items-center flex-col bg-gradient-radial from-[#121212] to-[#000000]">
      <RouterProvider router={router} />
    </main>
  );
}
