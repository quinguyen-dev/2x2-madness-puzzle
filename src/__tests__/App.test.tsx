import { expect, describe, it } from "vitest";
import { render } from "@testing-library/react";
import App from "../App";

describe("[App Component] Rendering", () => {
  it("should render no children", () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  });
});
