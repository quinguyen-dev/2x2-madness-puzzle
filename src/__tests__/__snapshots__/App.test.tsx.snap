// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[App Component] Rendering > should render no children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="h-screen flex justify-center items-center flex-col bg-gradient-radial from-[#121212] to-[#000000]"
      >
        <div
          class="text-white"
        >
          <h1
            class="text-6xl font-black select-none"
          >
            2x2 Madness
          </h1>
          <div
            class="flex flex-col space-y-2 mt-6 text-black"
          >
            <a
              class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
              href="/game"
            >
              <div
                class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
              />
              <span
                class="relative pl-4 group-hover:text-black group-hover:font-bold"
              >
                [
                4
                x
                4
                ]
                 
                E
                a
                s
                y
              </span>
            </a>
            <a
              class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
              href="/game"
            >
              <div
                class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
              />
              <span
                class="relative pl-4 group-hover:text-black group-hover:font-bold"
              >
                [
                5
                x
                5
                ]
                 
                M
                e
                d
                i
                u
                m
              </span>
            </a>
            <a
              class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
              href="/game"
            >
              <div
                class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
              />
              <span
                class="relative pl-4 group-hover:text-black group-hover:font-bold"
              >
                [
                6
                x
                6
                ]
                 
                H
                a
                r
                d
              </span>
            </a>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="h-screen flex justify-center items-center flex-col bg-gradient-radial from-[#121212] to-[#000000]"
    >
      <div
        class="text-white"
      >
        <h1
          class="text-6xl font-black select-none"
        >
          2x2 Madness
        </h1>
        <div
          class="flex flex-col space-y-2 mt-6 text-black"
        >
          <a
            class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
            href="/game"
          >
            <div
              class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
            />
            <span
              class="relative pl-4 group-hover:text-black group-hover:font-bold"
            >
              [
              4
              x
              4
              ]
               
              E
              a
              s
              y
            </span>
          </a>
          <a
            class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
            href="/game"
          >
            <div
              class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
            />
            <span
              class="relative pl-4 group-hover:text-black group-hover:font-bold"
            >
              [
              5
              x
              5
              ]
               
              M
              e
              d
              i
              u
              m
            </span>
          </a>
          <a
            class="text-white h-12 border-2 border-white flex items-center p-0 relative group cursor-clickable"
            href="/game"
          >
            <div
              class="absolute w-0 bg-white inset-0 transition-all duration-[250ms] group-hover:w-full"
            />
            <span
              class="relative pl-4 group-hover:text-black group-hover:font-bold"
            >
              [
              6
              x
              6
              ]
               
              H
              a
              r
              d
            </span>
          </a>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
